File: repair_schema.js
const fs = require('fs');
const data = JSON.parse(fs.readFileSync('appwrite.json', 'utf8'));

for (const table of data.tables || []) {
  if (table.$id === 'profiles') {
    const cols = table.columns;
    const keys = new Set(cols.map(c => c.key));
    
    const toAdd = [
        { key: 'current_voltz', type: 'integer', required: false, array: false, default: 0 },
        { key: 'visibility_boost_level', type: 'integer', required: false, array: false, default: 0 },
        { key: 'visibility_boost_expires_at', type: 'datetime', required: false, array: false, default: null, format: '' },
        { key: 'stripe_customer_id', type: 'varchar', required: false, array: false, size: 256, default: null, encrypt: false },
        { key: 'stripe_subscription_id', type: 'varchar', required: false, array: false, size: 256, default: null, encrypt: false }
    ];
    
    for (const c of toAdd) {
      if (!keys.has(c.key)) {
        cols.push(c);
        console.log('Added ' + c.key + ' to Profiles');
      }
    }
  }
  
  if (table.$id === 'voltz_ledger') {
    const cols = table.columns;
    const keys = new Set(cols.map(c => c.key));
    
    if (!keys.has('stripe_session_id')) {
      cols.push({ key: 'stripe_session_id', type: 'varchar', required: false, array: false, size: 256, default: null, encrypt: false });
      console.log('Added stripe_session_id to Voltz Ledger');
    }
    
    for (const col of cols) {
      if (col.key === 'profile') {
        col.required = false;
        console.log('Set voltz_ledger.profile to required=False');
      }
    }
  }

  if (table.$id === 'relationship_events') {
    for (const col of table.columns) {
      if (col.key === 'connection') {
        col.required = false;
        console.log('Set relationship_events.connection to required=False');
      }
    }
  }
}

fs.writeFileSync('appwrite.json', JSON.stringify(data, null, 4));
console.log('Saved appwrite.json');

File: repair_schema.py
import json

with open('appwrite.json', 'r') as f:
    data = json.load(f)

for table in data.get('tables', []):
    if table.get('$id') == 'profiles':
        cols = table.get('columns', [])
        keys = set(c['key'] for c in cols)
        
        to_add = [
            {'key': 'current_voltz', 'type': 'integer', 'required': False, 'array': False, 'default': 0},
            {'key': 'visibility_boost_level', 'type': 'integer', 'required': False, 'array': False, 'default': 0},
            {'key': 'visibility_boost_expires_at', 'type': 'datetime', 'required': False, 'array': False, 'default': None, 'format': ''},
            {'key': 'stripe_customer_id', 'type': 'varchar', 'required': False, 'array': False, 'size': 256, 'default': None, 'encrypt': False},
            {'key': 'stripe_subscription_id', 'type': 'varchar', 'required': False, 'array': False, 'size': 256, 'default': None, 'encrypt': False}
        ]
        
        for c in to_add:
            if c['key'] not in keys:
                cols.append(c)
                print(f"Added {c['key']} to Profiles")
                
    if table.get('$id') == 'voltz_ledger':
        cols = table.get('columns', [])
        keys = set(c['key'] for c in cols)
        
        if 'stripe_session_id' not in keys:
            cols.append({'key': 'stripe_session_id', 'type': 'varchar', 'required': False, 'array': False, 'size': 256, 'default': None, 'encrypt': False})
            print('Added stripe_session_id to Voltz Ledger')
            
        for col in cols:
            if col.get('key') == 'profile':
                col['required'] = False
                print('Set voltz_ledger.profile to required=False')

    if table.get('$id') == 'relationship_events':
        for col in table.get('columns', []):
            if col.get('key') == 'connection':
                col['required'] = False
                print('Set relationship_events.connection to required=False')

with open('appwrite.json', 'w') as f:
    json.dump(data, f, indent=4)
print('Saved appwrite.json')
